### PR TITLE
Inline verbatim

### LINKED ISSUE #7: In the link, the `` is not being parsed correctly, or there is a problem with my understanding.
```norg
    {alink}[`description in the symbol`]
```
the content in tree-sitter now be parsed as 
```
   content: (paragraph ; [16, 4] - [17, 0]
      (paragraph_segment ; [16, 4] - [16, 40]
        (link ; [16, 4] - [16, 40]
          (link_location ; [16, 4] - [16, 11]
            type: (link_target_url) ; [16, 5] - [16, 5]
            text: (paragraph)) ; [16, 5] - [16, 10]
          (link_description ; [16, 11] - [16, 40]
            text: (paragraph ; [16, 12] - [16, 39]
              (verbatim)))))) ; [16, 12] - [16, 39]
```
but in rust-norg , it was parsed as
```
Paragraph([Link { filepath: None, targets: [Url("alink")], description: Some([Token(Text("`description")), Token(Whitespace), Token(Text("in")), Token(Whitespace), Token(Text("the")), Token(Whitespace), Token(Text("sympol`"))]) }])
```

if I remove the ``
```norg
    {alink}[description in the symbol]
```
then in tree-sitter it is
```
  content: (paragraph ; [16, 4] - [17, 0]
      (paragraph_segment ; [16, 4] - [16, 38]
        (link ; [16, 4] - [16, 38]
          (link_location ; [16, 4] - [16, 11]
            type: (link_target_url) ; [16, 5] - [16, 5]
            text: (paragraph)) ; [16, 5] - [16, 10]
          (link_description ; [16, 11] - [16, 38]
            text: (paragraph))))) ; [16, 12] - [16, 37]
```
the different is  `(verbatim)`

I am not sure if this behavior is within expectations.

sorry for my poor English, but Neorg is really a amazing project

### PR DESCRIPTION
**Pull request by awegsche**
Adds inline verbatim (``normal text `inline verbatim` ``).

Fixes #7 

Adds some tests for it:

`` `inline verbatim` ``
`` {:file:}[inside `anchor`]``
`` ` *markup* /inside /-verbatim-/ ` ``

## Note
- maybe extend tests for missing cases

### CODE REVIEW

File: src/snapshots/rust_norg__tests__inline_verbatim.snap.new
Comment (NTBBloodbath) [inline src/snapshots/rust_norg__tests__inline_verbatim.snap.new:1]: thought: I think this file shouldn't end with a `.new` extension. However, I'm not entirely sure about how snapshots work so perhaps the tests suite doesn't care about it?

cc @vhyrro

Comment (awegsche) [reply src/snapshots/rust_norg__tests__inline_verbatim.snap.new:1]: I'm also not familiar with these snapshot tests. I just checked in the file as-is. 

I guess it needs to be reviewed as part of the PR, I don't know at which point the renaming should happen (before the check-in makes sense, actually).

Comment (awegsche) [reply src/snapshots/rust_norg__tests__inline_verbatim.snap.new:1]: renamed the file

Comment (NTBBloodbath) [reply src/snapshots/rust_norg__tests__inline_verbatim.snap.new:1]: Yeah, it seems that was what was causing the CI to fail. Everything looks fine to me :P

**Review by NTBBloodbath — APPROVED** [review]

**Comment by benlubas** [comment]
FYI, new snapshots need to be approved, you can approve them with `cargo insta review` (insta is the snapshot testing framework that's used)

I can add a contributing.md with this information

**Pull request merged**